### PR TITLE
Sc/use deployutils in tests

### LINKED ISSUE #12639: EVM Engineering: replace new ProxyAdmin and new Proxy in tests with DeployUtils
`new ProxyAdmin` and `new Proxy` are used in a number of places in the tests. Relatively straightforward task, replace these with calls to `DeployUtils.create1` and replace the imports with interface imports. 

Could also be nice to replace some `new <Contract>` calls in other places too:

- `WETH98.t.sol` - `new WETH98`
- `MintManager.t.sol` - `new GovernanceToken`, `new MintManager` 
- `SystemConfig.t.sol` - `new SystemConfig`
- `DelayedVetoable.t.sol` - `new DelayedVetoable`
- `SuperchainConfig.t.sol` - `new SuperchainConfig`
- `OptimismSuperchainERC20.t.sol` - `new OptimismSuperchainERC20`
- `SequencerFeeVault.t.sol` - `new SequencerFeeVault`
- `DeployerWhitelist.t.sol` - `new DeployerWhitelist`
- `L1BlockNumber.t.sol` - `new L1Block` (replace with `vm.getDeployedCode`), `new L1BlockNumber`
- `ResolvedDelegateProxy.t.sol` - `new AddressManager`, `new ResolvedDelegateProxy`
- `DeputyGuardianModule.t.sol` - `new DeputyGuardianModule`
- `BenchmarkTest.t.sol` - `new L1BlockInterop`
- `InitializableOZv5.t.sol` - `new OptimismSuperchainERC20`

### PR DESCRIPTION
**Pull request by AmadiMichael**
Fixes https://github.com/ethereum-optimism/optimism/issues/12639

Note: `WETH98.t.sol - new WETH98` can only be implemented after https://github.com/ethereum-optimism/optimism/pull/12648 is merged. The latter PR is where pseudo-constructors are added for IWETH{98}

### CODE REVIEW

File: packages/contracts-bedrock/test/L1/DelayedVetoable.t.sol [hunk 1]
```diff
@@ -4,6 +4,7 @@ pragma solidity 0.8.15;
 import { Test } from "forge-std/Test.sol";
 import { DelayedVetoable } from "src/L1/DelayedVetoable.sol";
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/DelayedVetoable.t.sol]: Should be able to get rid of this import now

File: packages/contracts-bedrock/test/L1/SuperchainConfig.t.sol [hunk 2]
```diff
@@ -4,12 +4,14 @@ pragma solidity 0.8.15;
 import { CommonTest } from "test/setup/CommonTest.sol";
 
 // Target contract dependencies
-import { Proxy } from "src/universal/Proxy.sol";
+import { IProxy } from "src/universal/interfaces/IProxy.sol";
 
 // Target contract
 import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/SuperchainConfig.t.sol]: Should be able to get rid of this import

File: packages/contracts-bedrock/test/L2/OptimismSuperchainERC20.t.sol [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.25;
+pragma solidity ^0.8.0;
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/OptimismSuperchainERC20.t.sol]: Why this diff?

File: packages/contracts-bedrock/test/invariants/SystemConfig.t.sol [hunk 4]
```diff
@@ -4,15 +4,26 @@ pragma solidity 0.8.15;
 import { Test } from "forge-std/Test.sol";
 import { SystemConfig } from "src/L1/SystemConfig.sol";
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/invariants/SystemConfig.t.sol]: Should be able to get rid of this import

File: packages/contracts-bedrock/test/governance/MintManager.t.sol [hunk 5]
```diff
@@ -23,10 +24,20 @@ contract MintManager_Initializer is CommonTest {
         super.setUp();
 
         vm.prank(owner);
-        gov = IGovernanceToken(address(new GovernanceToken()));
+        gov = IGovernanceToken(
+            DeployUtils.create1({
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/governance/MintManager.t.sol:24]: Should be able to get rid of the imports for `GovernanceToken` and `MintManager`

File: packages/contracts-bedrock/test/L2/OptimismSuperchainERC20.t.sol (hunk 3 above)
Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L2/OptimismSuperchainERC20.t.sol]: From when i was trying to get it to compile. Not necessary as i changed it in DeployUtils

File: packages/contracts-bedrock/test/universal/ProxyAdmin.t.sol [hunk 6]
```diff
@@ -6,33 +6,46 @@ import { Test } from "forge-std/Test.sol";
 import { SimpleStorage } from "test/universal/Proxy.t.sol";
 
 // Contracts
-import { Proxy } from "src/universal/Proxy.sol";
-import { ProxyAdmin } from "src/universal/ProxyAdmin.sol";
 import { AddressManager } from "src/legacy/AddressManager.sol";
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/universal/ProxyAdmin.t.sol]: Can we also apply `DeployUtils` to these three?

File: packages/contracts-bedrock/test/universal/ProxyAdmin.t.sol [hunk 7]
```diff
@@ -6,46 +6,75 @@ import { Test } from "forge-std/Test.sol";
 import { SimpleStorage } from "test/universal/Proxy.t.sol";
 
 // Contracts
-import { Proxy } from "src/universal/Proxy.sol";
-import { ProxyAdmin } from "src/universal/ProxyAdmin.sol";
-import { AddressManager } from "src/legacy/AddressManager.sol";
-import { L1ChugSplashProxy } from "src/legacy/L1ChugSplashProxy.sol";
-import { ResolvedDelegateProxy } from "src/legacy/ResolvedDelegateProxy.sol";
+import { IAddressManager } from "src/legacy/interfaces/IAddressManager.sol";
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/universal/ProxyAdmin.t.sol:9]: One last nit, these should be moved into `Interfaces` section below

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**